ui/handler.go

```diff
@@ -24,28 +24,43 @@ func Handler(devPort int, client kclient.Client) http.Handler {
 		lock:   new(sync.RWMutex),
 	}
 
-	if devPort == 0 {
-		return server
-	}
-	return &httputil.ReverseProxy{
-		Director: func(r *http.Request) {
-			r.URL.Scheme = "http"
-			if strings.HasPrefix(r.URL.Path, "/admin") {
-				r.URL.Host = fmt.Sprintf("localhost:%d", devPort)
-			} else {
-				r.URL.Host = fmt.Sprintf("localhost:%d", devPort+1)
-			}
-		},
+	if devPort != 0 {
+		server.rp = &httputil.ReverseProxy{
+			Director: func(r *http.Request) {
+				r.URL.Scheme = "http"
+				if strings.HasPrefix(r.URL.Path, "/admin") {
+					r.URL.Host = fmt.Sprintf("localhost:%d", devPort)
+				} else {
+					r.URL.Host = fmt.Sprintf("localhost:%d", devPort+1)
+				}
+			},
+		}
```

Comment:
I moved the reverse proxy into the uiServer object, rather than returning it as an alternative to the uiServer. That way we can have one unified check for the oauth-related query parameters, before serving the request with either the reverse proxy (for the `make dev` setup) or the normal uiServer.